build/Gruntfile-ci.coffee

```diff
@@ -0,0 +1,295 @@
+            dest: path.join(appDir, jsFile)
+          })
+
+  grunt.initConfig
+    pkg: grunt.file.readJSON('package.json')
+
+    atom: {
+      appName, channel, metadata, disableAutoUpdate,
+      appFileName, apmFileName,
+      appDir, buildDir, contentsDir, installDir, shellAppDir, symbolsDir,
+    }
+
+    docsOutputDir: 'docs/output'
+
+    babel: babelConfig
+
+    coffee: coffeeConfig
+
+    less: lessConfig
+
+    'prebuild-less': prebuildLessConfig
+
+    cson: csonConfig
+
+    peg: pegConfig
+
+    coffeelint:
+      options:
+        configFile: 'coffeelint.json'
+      src: [
+        'dot-atom/**/*.coffee'
+        'exports/**/*.coffee'
+        'src/**/*.coffee'
+      ]
+      build: [
+        'build/tasks/**/*.coffee'
+        'build/Gruntfile.coffee'
+      ]
+      test: [
+        'spec/*.coffee'
+      ]
+
+    standard:
+      src: [
+        'src/**/*.js'
+        'static/*.js'
+      ]
+
+    csslint:
+      options:
+        'adjoining-classes': false
+        'duplicate-background-images': false
+        'box-model': false
+        'box-sizing': false
+        'bulletproof-font-face': false
+        'compatible-vendor-prefixes': false
+        'display-property-grouping': false
+        'fallback-colors': false
+        'font-sizes': false
+        'gradients': false
+        'ids': false
+        'important': false
+        'known-properties': false
+        'outline-none': false
+        'overqualified-elements': false
+        'qualified-headings': false
+        'unique-headings': false
+        'universal-selector': false
+        'vendor-prefix': false
+      src: [
+        'static/**/*.css'
+      ]
+
+    lesslint:
+      src: [
+        'static/**/*.less'
+      ]
+
+    'download-electron':
+      version: packageJson.electronVersion
+      outputDir: 'electron'
+      downloadDir: electronDownloadDir
+      rebuild: true  # rebuild native modules after electron is updated
+      token: process.env.ATOM_ACCESS_TOKEN
+
+    'create-windows-installer':
+      installer:
+        appDirectory: shellAppDir
+        outputDirectory: path.join(buildDir, 'installer_' + process.arch)
+        authors: 'GitHub Inc.'
+        loadingGif: path.resolve(__dirname, '..', 'resources', 'win', 'loading.gif')
+        iconUrl: "https://raw.githubusercontent.com/atom/atom/master/resources/app-icons/#{channel}/atom.ico"
+        setupIcon: path.resolve(__dirname, '..', 'resources', 'app-icons', channel, 'atom.ico')
+
+    shell:
+      'kill-atom':
+        command: killCommand
+        options:
+          stdout: false
+          stderr: false
+          failOnError: false
+
+  grunt.registerTask('compile', ['babel', 'coffee', 'prebuild-less', 'cson', 'peg'])
+  grunt.registerTask('lint', ['standard', 'coffeelint', 'csslint', 'lesslint'])
+  grunt.registerTask('test', ['shell:kill-atom', 'run-specs'])
+
+  ciTasks = ['output-disk-space', 'download-electron', 'download-electron-chromedriver', 'build']
```

Comment:
`output-disk-space` should be disabled for at least Travis, and maybe Appveyor.  It currently doesn't output anything on Travis.